packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/sysi.c

```diff
@@ -0,0 +1,348 @@
+
+#include "x86_64_delta_ak7448_int.h"
+#include "x86_64_delta_ak7448_log.h"
+#include "platform_lib.h"
+
+const char*
+onlp_sysi_platform_get(void)
+{
+    return "x86-64-delta-ak7448-r0";
+}
+
+int
+onlp_sysi_init(void)
+{
+    return ONLP_STATUS_OK;
+}
+
+/******************* Utility Function *****************************************/
+int
+decide_percentage(int *percentage, int temper)
+{
+    int level;
+
+    if(temper <= 50)
+    {
+        *percentage = 50;
+        level = 1;
+    }
+    else if(temper > 50 && temper <= 55)
+    {
+        *percentage = 58;
+        level = 2;
+    }
+    else if(temper > 55 && temper <= 60)
+    {
+        *percentage = 65;
+        level = 3;
+    }
+    else if(temper > 60 && temper <= 65)
+    {
+        *percentage = 80;
+        level = 4;
+    }
+     else if(temper > 65)
+    {
+        *percentage = 100;
+        level = 5;
+    }
+    else
+    {
+        *percentage = 100;
+        level = 6;
+    }
+  
+    return level;
+}
+/******************************************************************************/
+
+
+int
+onlp_sysi_onie_data_get(uint8_t** data, int* size)
+{
+    uint8_t* rdata = aim_zmalloc(256);
+
+    if(onlp_file_read(rdata, 256, size, IDPROM_PATH) == ONLP_STATUS_OK) {
+        if(*size == 256) {
+            *data = rdata;
+            return ONLP_STATUS_OK;
+        }
+    }
+
+    aim_free(rdata);
+    *size = 0;
+    return ONLP_STATUS_E_UNSUPPORTED;
+}
+
+void
+onlp_sysi_onie_data_free(uint8_t* data)
+{
+    aim_free(data);
+}
+
+int
+onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
+{
+    int sys_cpld_version = 0, cpld_a_version = 0, cpld_b_version = 0;
+
+    sys_cpld_version = dni_lock_cpld_read_attribute(SYS_CPLD_PATH,SYS_VERSION_REG);
+    cpld_a_version = dni_lock_cpld_read_attribute(CPLD_A_PATH,CPLD_A_VERSION_REG);
+    cpld_b_version = (dni_lock_cpld_read_attribute(CPLD_B_PATH,CPLD_B_VERSION_REG) & 0x7);
+
+    pi->cpld_versions = aim_fstrdup("SYSTEM-CPLD = %d, CPLD-A = %d, CPLD-B = %d", sys_cpld_version, cpld_a_version, cpld_b_version);
+
+    return ONLP_STATUS_OK;
+}
+
+void
+onlp_sysi_platform_info_free(onlp_platform_info_t* pi)
+{
+    aim_free(pi->cpld_versions);
+}
+
+int
+onlp_sysi_oids_get(onlp_oid_t* table, int max)
+{
+    int i;
+    onlp_oid_t* e = table;
+    memset(table, 0, max*sizeof(onlp_oid_t));
+
+    /* 7 Thermal sensors on the chassis */
+    for (i = 1; i <= NUM_OF_THERMAL_ON_BOARDS; i++)
+    {
+        *e++ = ONLP_THERMAL_ID_CREATE(i);
+    }
+
+    /* 6 LEDs on the chassis */
+    for (i = 1; i <= NUM_OF_LED_ON_BOARDS; i++)
+    {
+        *e++ = ONLP_LED_ID_CREATE(i);
+    }
+
+    /* 7 Fans on the chassis */
+    for (i = 1; i <= NUM_OF_FAN_ON_FAN_BOARD; i++)
+    {
+        *e++ = ONLP_FAN_ID_CREATE(i);
+    }
+
+    /* 1 PSUs on the chassis */
+    for (i = 1; i <= NUM_OF_PSU_ON_PSU_BOARD; i++)
+    {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+int
+onlp_sysi_platform_manage_fans(void)
```

Comment:
The API lock is not held while executing the platform fan and led management functions. 
The fani, thermali, and ledi implementations cannot be called here. Please use the normal fan, thermal, and led calls.

Reply:
Hi @jnealtowns:
Thank for your comment but we have some question.
 1. The ONLPI Driver APIs are called by ONLP Application APIs and we see there are API lock function in ONLP 
     Application  APIs. Why do we need to add the lock function in ONLPI Driver APIs?
 2.  We do not know about "normal fan. thermal and led calls" exactly. You means to call the ONLP Application APIs in 
        sysi.c, is it appropriate?

Reply:
You are correct. All ONLPI implementations are protected against concurrency by the upper ONLP layer through shared mutex locks. 

The *only* ones which *aren't* protected are the onlp_sysi_platform_manage_* functions. They are poorly named to reflect this.

Reply:
I will update the onlp_sysi_platform_manage_* functions which reflect the protection while calling ONLPI.